test_conformance/buffers/test_buffer_fill.c

```diff
@@ -779,8 +779,6 @@ int test_buffer_fill_struct( cl_device_id deviceID, cl_context context, cl_comma
                 buffers[0] = clCreateBuffer(context, flag_set[src_flag_id],  ptrSize * num_elements, NULL, &err);
             if ( err ){
                 print_error(err, " clCreateBuffer failed\n" );
-                clReleaseEvent( event[0] );
```

Comment:
What is the purpose of removing these clReleaseEvent functions in this file ?

Reply:
+1, this looks odd.

Reply:
The negative flow of this and a few other tests is faulty (it has been copy pasted). The resources being freed are actually resources that never got allocated at the point of this check. Since the generate sources only mode returns an error (the compilation doesn't succeed), I actually trigger this negative flow and get a segmentation fault when running the conformance in this use case. Please have a look at the entire code base and let me know if I missed something.

Reply:
Ok I see the issue and agree with the your proposal. However this should be reported as separate Issue and fix.